AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/MutationRetryNotifier.swift

```diff
@@ -0,0 +1,117 @@
+//
+// Copyright 2018-2019 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+import AWSPluginsCore
+import Combine
+
+@available(iOS 13.0, *)
+final class MutationRetryNotifier {
+    private var nextSyncTimer: DispatchSourceTimer?
+    private var retryAttemptNumber: Int
+
+    private var handlerQueue: DispatchQueue = DispatchQueue.global(qos: .utility)
+    var retryMutationCallback: () -> Void
+    private var reachabilitySubscription: Subscription?
+
+    init(retryAttemptNumber: Int,
+         advice: RequestRetryAdvice,
+         retryMutationCallback: @escaping () -> Void) {
+        self.retryAttemptNumber = retryAttemptNumber
+        self.retryMutationCallback = retryMutationCallback
+
+        NetworkReachabilityNotifier.shared?.publisher().subscribe(self)
+
+        let deadline = DispatchTime.now() + advice.retryInterval!
```

Comment:
Let's set up a sane default rather than force-unwrapping

Reply:
Sane defaults sounds good.  Our max retry duration is about 5 minutes.. somehow 2.5 minutes seems too long, how about 60 seconds?